src/ext/NetFx/wixext/NetFxCompiler.cs

```diff
@@ -325,5 +361,184 @@ private void ParseNativeImageElement(Intermediate intermediate, IntermediateSect
                 });
             }
         }
+
+        /// <summary>
+        /// Parses a DotNetCompatibilityCheck element.
+        /// </summary>
+        /// <param name="element">The element to parse.</param>
+        private void ParseDotNetCompatibilityCheckElement(Intermediate intermediate, IntermediateSection section, XElement element)
+        {
+            var sourceLineNumbers = this.ParseHelper.GetSourceLineNumbers(element);
+            Identifier id = null;
+            string variable = null;
+            string condition = null;
+            string after = null;
+            string runtimeType = null;
+            string platform = null;
+            string version = null;
+            string rollForward = "Minor";
+
+            foreach (var attrib in element.Attributes())
+            {
+                if (String.IsNullOrEmpty(attrib.Name.NamespaceName) || this.Namespace == attrib.Name.Namespace)
+                {
+                    switch (attrib.Name.LocalName)
+                    {
+                        case "Id":
+                            id = this.ParseHelper.GetAttributeIdentifier(sourceLineNumbers, attrib);
+                            break;
+                        case "Variable":
+                            variable = this.ParseHelper.GetAttributeBundleVariableNameValue(sourceLineNumbers, attrib);
+                            break;
+                        case "Condition":
+                            condition = this.ParseHelper.GetAttributeValue(sourceLineNumbers, attrib);
+                            break;
+                        case "After":
+                            after = this.ParseHelper.GetAttributeValue(sourceLineNumbers, attrib);
+                            break;
+                        case "RuntimeType":
+                            runtimeType = this.ParseHelper.GetAttributeValue(sourceLineNumbers, attrib);
+                            switch (runtimeType.ToLower())
+                            {
+                                case "aspnet":
+                                    runtimeType = "Microsoft.AspNetCore.App";
+                                    break;
+                                case "desktop":
+                                    runtimeType = "Microsoft.WindowsDesktop.App";
+                                    break;
+                                case "core":
+                                    runtimeType = "Microsoft.NETCore.App";
+                                    break;
+                                default:
+                                    this.Messaging.Write(ErrorMessages.IllegalAttributeValueWithLegalList(sourceLineNumbers, element.Name.LocalName, attrib.Name.LocalName, runtimeType, "aspnet, desktop and core"));
```

Comment:
We should probably find a way to share code for these attributes that are shared with `DotNetCoreSearch`. If not, then the last parameter should be split up into `"aspnet", "desktop", "core"`.

Reply:
As I saw, you're using enumerations for runtime type and platform. I could move to using that approach (plus add enumeration for roll forward policy) and then in CA instead of just reading values and passing them to NetCoreCheck.exe, I'd need to have map them back to values recognized by NetCoreChec.exe. In the end it comes down to which part of code do we want to have cleaner, so let me know your preference and I'll implement it that way.

Reply:
Maybe we'll try to get them to share code after this PR.